ethereum/core/src/main/java/org/hyperledger/besu/ethereum/privacy/PrivateTransactionReceipt.java

```diff
@@ -64,12 +62,22 @@ public PrivateTransactionReceipt(
 
   public PrivateTransactionReceipt(final TransactionProcessor.Result result) {
     this(
-        result.getStatus() == PrivateTransactionProcessor.Result.Status.SUCCESSFUL ? 1 : 0,
+        getStatusCode(result.getStatus()),
         result.getLogs(),
         result.getOutput(),
         result.getRevertReason());
   }
 
+  private static int getStatusCode(final TransactionProcessor.Result.Status result) {
+    if (result.equals(PrivateTransactionProcessor.Result.Status.SUCCESSFUL)) {
```

Comment:
NIT: we could use a switch statement here, couldn't we?

Reply:
Also, we could introduce constants `STATUS_SUCCESSFUL = 1`, `STATUS_FAILED = 0` and `STATUS_INVALID = 2` to highlight the mapping btw status and the numbers.